content/docs/guides/auth/ALTS.md

```diff
@@ -0,0 +1,246 @@
+---
```

Comment:
So I think this should probably be broken down into two pages.
1. A general purpose guide describing how to use the ALTS feature in gRPC (platform agnostic)
2. A new blog post describing how to use the ALTS feature inside of Google Cloud Platform

Given that ALTS has an associated white-papper, I expect implementations to be cropping up any time. In a cursory search, I was able to find a relevant thread someone started on [Istio](https://discuss.istio.io/t/what-does-the-alts-handshaker-do/6072). Since we can expect other implementations to crop up, it would be helpful to decouple Google Cloud Platform specifics from the core feature guide.

@thisisnotapril and I were talking earlier about good content for blog posts. I think provider-specific examples are great candidates for this.

Curious what other peoples thoughts are on this.

Reply:
While the Istio example is referencing the client-side, I could see something like Hashicorp Vault supporting the server-side of ALTS.

Reply:
Unfortunately, gRPC ALTS only works for GCP. It does not work on-prem or on other Clouds.

Reply:
Sure. What I'm suggesting is that it's only a matter of time until we see other implementations. Between the white paper and the client-side implementation, there's enough context to piece together alternative backend implementations.

Reply:
Thanks @mjpitz for suggestion. I split the guide into two pages: one generic guide, the other is Google Cloud Platform page. I also made a note that ALTS is currently available on GCP. 

Technically, if someone writes an ALTS handshaker service, it can run outside GCP. It is out of scope of this guide.

Reply:
Oh I completely agree that is outside the scope of this change. It's just something I thought we should consider as we developed content around it.

Reply:
Do you have specific suggestion on developing content? For ALTS on GCP or ALTS outside GCP?

Reply:
So the big thing that I try to consider is separating documentation of a given feature from implementation or use. In this case, ALTS (feature) and GCP/GKE (use case). This is [generally?] a good practice to follow when there are pluggable interfaces or configuration that allows backend implementation details to be changed out.

Let us consider the previous version where everything was in a single document. If someone like Vault came along and implemented an ALTS handshake service, its reasonable to expect an end-user to request documentation around this integration. We would then need to update to the ALTS documentation to detail both GCP's and Vaults support or remove the provider specifics entirely.

As the content currently stands, we get everything we wanted from the original (documentation of ALTS feature and it's integration with GCP/GKE) while being able to decouple provider-specific details from the core guide. I feel like the blog posts are a great way to showcase integrations with different providers, while keeping the core documentation agnostic.

Reply:
Thanks for suggestion!

As for blog post, I got my leadership approval for ALTS user guide, but not blog post yet. I hope we can publish user guide first, so that user can start using this feature.

Reply:
Let me remove GCP specific information from this guide. I agree it is not suitable in this guide.